### PR TITLE
Tp 8693ur398 card layout

### PR DESCRIPTION
**Pull request by HeySleare**

### CODE REVIEW

File: src/components/ProductCard/ProductCard.tsx [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
       <img
-        src="../img/phones/apple-iphone-14-pro/spaceblack/00.webp"
+        src={`../${phone.images[0]}`}
         alt="Iphone 14 Midnight"
         className="card__image"
       />
 
       <h2 className="card__title">
-        Apple iPhone 14 Pro 128GB Deep Purple (MQ0G3)
+        {phone.name} (MQ023)
```
Comment (sasholk) [inline src/components/ProductCard/ProductCard.tsx:20]:
```suggestion
        {phone.name}
```

leave only `phone.name`, cuz we don't have info about models for now

File: src/components/ProductCard/ProductCard.tsx [hunk 2]
```diff
@@ -1,36 +1,41 @@
...
+import './productcard.scss';
+import { Phone } from '../../types/Phone';
 
-export const CardPhone: React.FC = () => {
+type Props = {
+  phone: Phone;
+};
+
+export const ProductCard: React.FC<Props> = ({ phone }) => {
   return (
```
Comment (sasholk) [inline src/components/ProductCard/ProductCard.tsx:11]:
```suggestion
export const ProductCard: React.FC<Props> = ({ phone }) => {
const {...} = phone // mb destructuring would be better 
  return (
```

File: src/components/ProductCard/productcard.scss [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+
 .card {
+  margin: 0 auto;
   display: flex;
   flex-direction: column;
   padding: 32px;
-  width: 272px;
+  width: $card-width;
   height: 506px;
+  border: 1px #E2E6E9 solid; // тимчасовий бордер, для того, щоб бачити рамки карток. ПОФІКСІТЬ НА НОРМАЛЬНИЙ БОРДЕР!
```
Comment (sasholk) [inline src/components/ProductCard/productcard.scss:10]: 🙏🙏🙏

File: src/pages/CatalogPage/CatalogPage.tsx [hunk 4]
```diff
@@ -1,14 +1,32 @@
 import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
 import { Pagination } from '../../components/Pagination';
+import * as getGoods from '../../api/products';
```
Comment (sasholk) [inline src/pages/CatalogPage/CatalogPage.tsx]: goods? mb better to name it more correctly

**Review by sasholk — CHANGES_REQUESTED** [review]

File: src/api/products.ts [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
 
-export const getTablets = () => {
+export const tablets = () => {
   return request(Product.Tablets);
 };
 
-export const getProducts = () => {
+export const products = () => {
   return request(Product.Products);
 };
```
Comment (artem8746) [inline src/api/products.ts]: change to getPhones, getProducts etc

File: src/components/CardLayout/CardLayout.tsx
Comment (artem8746) [inline src/components/CardLayout/CardLayout.tsx:1]: ProductTable

File: src/pages/CatalogPage/CatalogPage.tsx [hunk 6]
```diff
@@ -1,14 +1,33 @@
...
 export const CatalogPage = () => {
   const { currentPageNumber } = useParams();
+  const [phones, setPhones] = useState<Phone[]>([]);
+  const [amountOfPages, setAmountOfPages] = useState(1);
+  const [itemPerPage] = useState(16);
+
+  useEffect(() => {
+    getPhones().then(data => {
+      setPhones(data);
+    });
```
Comment (artem8746) [inline src/pages/CatalogPage/CatalogPage.tsx:16]:
```suggestion
    getPhones().then(setPhones);
```

File: src/pages/CatalogPage/CatalogPage.tsx [hunk 7]
```diff
@@ -1,14 +1,33 @@
...
+
+  }, []);
+
+  useEffect(() => {
+    setAmountOfPages( Math.ceil(phones.length / itemPerPage));
+  }, [phones]);
```
Comment (artem8746) [inline src/pages/CatalogPage/CatalogPage.tsx]:
```suggestion

```

**Review by artem8746 — APPROVED** [review]
💞💥💦💦💦💦

**Pull request merged**